ehrapy/tools/cohort_tracking/_cohort_tracker.py

```diff
@@ -0,0 +1,332 @@
+                adjusted_colors = [((color[0] + hue_shift) % 1, color[1], color[2]) for color in colors]
+
+                # for categoricals, plot multiple bars
+                if _cols in self.categorical:
+                    for i, value in enumerate(data):
+                        ax.barh(pos, value, left=cumwidth, color=adjusted_colors[i], height=0.7)
+
+                        if value > 5:
+                            # Add proportion numbers to the bars
+                            width = value
+                            ax.text(
+                                cumwidth + width / 2,
+                                pos,
+                                f"{value:.1f}",
+                                ha="center",
+                                va="center",
+                                color="white",
+                                fontweight="bold",
+                            )
+
+                        ax.set_yticks([])
+                        ax.set_xticks([])
+                        cumwidth += value
+                        legend_labels.append(data.index[i])
+
+                # for numericals, plot a single bar
+                else:
+                    ax.barh(pos, 100, left=cumwidth, color=adjusted_colors[0], height=0.8)
+                    ax.text(
+                        100 / 2,
+                        pos,
+                        data[0],
+                        ha="center",
+                        va="center",
+                        color="white",
+                        fontweight="bold",
+                    )
+                    legend_labels.append(_cols)
+
+            # Set y-axis labels
+            if set_axis_labels:
+                ax.set_yticks(
+                    range(len(self.track.keys()))
+                )  # Set ticks at positions corresponding to the number of columns
+                ax.set_yticklabels(self.track.keys())  # Set y-axis labels to the column names
+
+            # makes the frames invisible
+            # for ax in axes:
+            #     ax.axis('off')
+
+        # Add legend
+        tot_legend_kwargs = {"loc": "best", "bbox_to_anchor": (1, 1)}
+        if legend_kwargs is not None:
+            tot_legend_kwargs.update(legend_kwargs)
+
+        plt.legend(legend_labels, **tot_legend_kwargs)
+
+        if save is not None:
+            if not isinstance(save, str):
+                raise ValueError("'save' must be a string.")
+            plt.savefig(
+                save,
+            )
+
+        if return_plot:
+            return fig, axes
+
+        else:
+            plt.tight_layout()
+            plt.show()
+
+    def plot_flowchart(self, save: str = None, return_plot: bool = True):
+        """Flowchart over the tracked steps.
+
+        Create a simple flowchart of data preparation steps tracked with `CohortTracker`.
+
+        Args:
+            save: If a string is provided, the plot will be saved to the path specified.
```

Comment:
This `save` behavior is different from the one scanpy uses (which is bullshit). I'm not sure whether we should have this. Is this interoperable with matplotlib in any way and can just be saved this way?

Reply:
Absolutely agree - using `return_figure=True` the user gets the figure object, and can do e.g. `fig.savefig("<name>.png", <other args>)` just as usual.

Asking for the matplotlib object should be the only way of saving the plot, I think now.

Reply:
Now go with a `show` argument: likely harmonized away in the future then